portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/fileinstall/FileInstall.testcase

```diff
@@ -121,6 +154,171 @@ definition {
+
+			LexiconEntry.gotoEntryMenuItemNoError(
+				menuItem = "Delete",
+				rowEntry = "/api/liferay/do");
+
+			AssertTextNotPresent(
+				locator1 = "//body",
+				value1 = "/api/liferay/do");
+		}
+
+		task ("When the default factory config is recreated via an OSGI config file") {
+			OSGiConfig.deployOSGiConfigFile(
+				osgiConfigFileName = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration~default.config",
+				osgiConfigs = "enabled=\"true\",hostsAllowed=\"127.0.0.1\",serviceAccessPolicyName=\"SYSTEM_USER_PASSWORD\",urlsIncludes=\"/api/liferay/do\"");
+		}
+
+		task ("Then the default factory configuration should be successfully re-added") {
+			Portlet.shutdownServer();
+
+			Portlet.startServer(deleteLiferayHome = "false");
+
+			SystemSettings.openSystemSettingsAdmin();
+
+			SystemSettings.gotoConfiguration(
+				configurationCategory = "API Authentication",
+				configurationName = "Tunnel Authentication",
+				configurationScope = "System Scope");
+
+			LexiconEntry.gotoEntry(rowEntry = "/api/liferay/do");
+
+			APIAuthentication.assertConfigInSystemSettings(
+				enabled = "true",
+				hostsAllowed = "127.0.0.1",
+				serviceAccessPolicyName = "SYSTEM_USER_PASSWORD",
+				urlsInclude = "/api/liferay/do");
+		}
+	}
+
+	@description = "Verify when an existing factory config is renamed, the old entry is removed and the new one is added."
+	@priority = "3"
+	test RenameFactoryConfig {
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration~original.config";
+		property osgi.module.configurations = "enabled=\"true\"${line.separator}hostsAllowed=\"127.0.0.1,SERVER_IP\"${line.separator}serviceAccessPolicyName=\"SYSTEM_USER_PASSWORD\"${line.separator}urlsIncludes=\"/api/liferay/to\"";
+		property test.name.skip.portal.instance = "FileInstall#RenameFactoryConfig";
+
+		task ("When a factory configuration is renamed") {
+			var configName = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration";
+			var liferayHome = PropsUtil.get("liferay.home.dir.name");
+
+			var configFileOriginal = "${liferayHome}/osgi/configs/${configName}~original.config";
+			var configFileNew = "${liferayHome}/osgi/configs/${configName}~new.config";
+
+			AntCommand(
+				locator1 = "build-test.xml",
+				value1 = "rename-file -Dfile.name=${configFileOriginal} -Dfile.rename=${configFileNew}");
+		}
+
+		task ("Then the old entry is removed from the database and the new one is added") {
+			Portlet.shutdownServer();
+
+			Portlet.startServer(deleteLiferayHome = "false");
+
+			var configName = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration";
+
+			var configTable = SQL.executeMySQLStatement(mysqlStatement = "SELECT * FROM lportal.Configuration_ WHERE configurationId LIKE '%${configName}~original%'");
```

Comment:
Why are we checking the database only now and not before? We should keep the assertions consistent if they are supposed to assert the same thing. For example: the UI technically just grabs the configuration from the database and displays it. So we are performing a duplicate check here. However if there is a reason for performing the duplicate check, then that is fine as well